korge/src/androidMain/kotlin/com/soywiz/korge/service/vibration/NativeVibration.kt

```diff
@@ -1,16 +1,46 @@
 package com.soywiz.korge.service.vibration
 
-import android.annotation.*
-import android.os.*
-import android.support.v4.content.ContextCompat.*
-import com.soywiz.klock.*
-import com.soywiz.korge.android.*
-import com.soywiz.korge.view.*
+import android.annotation.SuppressLint
+import android.os.VibrationEffect
+import android.os.Vibrator
+import android.support.v4.content.ContextCompat.getSystemService
+import com.soywiz.klock.TimeSpan
+import com.soywiz.korge.android.androidActivity
+import com.soywiz.korge.view.Views
 
-actual class NativeVibration actual constructor(val views: Views) {
+actual class NativeVibration actual constructor(views: Views) {
+
+    companion object {
+        private const val NO_REPEAT = -1
+    }
+
+    private val vibrator = getSystemService(views.androidActivity, Vibrator::class.java)
+
+    /**
+     * @param timings list of alternating ON-OFF durations in milliseconds. Staring with ON.
+     * @param amplitudes list of intensities of the vibration (0-255)
+     */
+    @ExperimentalUnsignedTypes
+    @SuppressLint("MissingPermission")
+    actual fun vibratePattern(timings: Array<TimeSpan>, amplitudes: Array<UInt>) {
+        val onOffTimings = (listOf(TimeSpan.NULL) + timings).map { it.millisecondsLong }.toLongArray()
+        if (amplitudes.size != onOffTimings.size) {
+            vibrator?.vibrate(VibrationEffect.createWaveform(onOffTimings, NO_REPEAT))
+        } else {
+            vibrator?.vibrate(
+                VibrationEffect.createWaveform(onOffTimings, amplitudes.toUIntArray().toIntArray(), NO_REPEAT)
+            )
+        }
+
+    }
+
+    /**
+     * @param time vibration duration in milliseconds
+     * @param amplitude intensity of the vibration (0-255)
+     */
+    @ExperimentalUnsignedTypes
     @SuppressLint("MissingPermission")
-    actual fun vibrate(pattern: Array<TimeSpan>) {
-        val v = getSystemService(views.androidActivity, Vibrator::class.java)
-        v?.vibrate(VibrationEffect.createWaveform(pattern.map { it.millisecondsLong }.toLongArray(), -1))
+    actual fun vibrate(time: TimeSpan, amplitude: UInt) {
```

Comment:
If it is a value between 0-255 why not use a UByte instead, or just an Int?
Any why not use a Double or Float with a value between 0.0 and 1.0?